src/snowflake/snowpark/_internal/analyzer/select_statement.py

```diff
@@ -149,6 +155,7 @@ def __init__(self, analyzer: "Analyzer") -> None:
         self._column_states: Optional[ColumnStateDict] = None
         self._snowflake_plan: Optional[SnowflakePlan] = None
         self.expr_to_alias = {}
+        self._api_calls = api_calls
```

Comment:
You need to copy this list, otherwise it will be the same list for multiple `Selectable` objects. If you look at `SnowflakePlan` you'll see that I always copy the list that is passed to the __init__ method to a new list.

Reply:
Previously `_api_calls` is copied when there is a SelectStatement API is called.
Changed the copy the the constructor.